sympy/functions/elementary/hyperbolic.py

```diff
@@ -1693,6 +1693,15 @@ def expansion_term(n, x, *previous_terms):
                 F = factorial(k) * n // 2 * n // 2
                 return -1 * R / F * x**n / 4
 
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):
+        from sympy.series.order import Order
+        arg = self.args[0].as_leading_term(x)
+
+        if x in arg.free_symbols and Order(1, x).contains(arg):
+            return -1*log(arg)
```

Comment:
It seems that this should be something like `return log(2) - log(c) - e*log(x)` when `arg` is `c*x**e`. If `log(2)` is not included, then the leading term of `log(x) + asech(x)` will not be correct. `log(c*x**e)` should be written in the form `log(c) + e*log(x)` so that the `log(x)` terms coming from different parts of a sum would combine without explicit simplifications.

Reply:
Thanks for pointing this out , I had completely overlooked this aspect but yeah surely what you've said if correct

Reply:
I was about to make a commit but it seems that this test would fail as it gives back log(2) rather than the output below when I make the above change , will have to look into this !
```
>>> limit(log(x)+asech(x),x,0, dir = '-')
log(2) + 2*I*pi
```

Reply:
log(x) + asech(x) = log(x) + log((1 + sqrt(1 - x**2))/x)
                      = log(1 + sqrt(1 - x**2))

is meromorphic (even holomorphic) at `x = 0` so its limit from any direction should be equal to its value `log(2)`.

Reply:
Ahh okay now I realize what led to my confusion ! The above returns True and below one returns false but yeah what you said is correct `log(2)` should be the output for the this
```
>>> (log(1 + sqrt(1 - x**2))).is_meromorphic(x, 0)
True
>>> (log(x) + asech(x)).is_meromorphic(x, 0)
>>>
```

Reply:
> asech(x)

@jksuom I have been having doubts in this recently , are we always allowed to use ` log` operations ( I don't think there any restrictions cause i've never heard of any)  
For eg
```
>>> asech(x).rewrite(log)
log(sqrt(-1 + 1/x)*sqrt(1 + 1/x) + 1/x)

>>> limit(asech(x).rewrite(log) + log(x), x, 0,'-')
log(2) + 2*I*pi

>>> limit(log(sqrt(-1 + 1/x)*sqrt(1 + 1/x) + 1/x) + log(x), x, 0,'-')
log(2) + 2*I*pi

>>> limit(log(x*sqrt(-1 + 1/x)*sqrt(1 + 1/x) + 1), x, 0,'-') # After performing log simplifcation 
log(2)
```

Reply:
I'll paste a set of results which I feel are the correct ones (after confirming with wolfram and other CAS)
Ignore the rewrite required in these, all results are correct but in sympy we could only get the correct results as of now on master using the rewrite ! But other than that wolfram agrees to all these answers !
```
>>> limit(log(x) + asech(x).rewrite(log), x,0, '-')
log(2) + 2*I*pi
>>> limit(log(x) + asech(x).rewrite(log), x,0, '+')
log(2)
>>> limit(-log(x) + asech(x).rewrite(log), x,0, '-')
oo
>>> limit(-log(x) + asech(x).rewrite(log), x,0, '+')
oo
>>>
>>>
>>> limit(log(x) + acsch(x).rewrite(log), x,0, '-')
-oo
>>> limit(log(x) + acsch(x).rewrite(log), x,0, '+')
log(2)
>>> limit(-log(x) + acsch(x).rewrite(log), x,0, '-')
-log(2) - I*pi
>>> limit(-log(x) + acsch(x).rewrite(log), x,0, '+')
oo
```
The best part is my first commit was giving all answers perfectly and after trying to consider the suggestion you pointed above now this is what we get 
```
>>> limit(log(x) + asech(x), x, 0, '-')
log(2)
>>> limit(log(x) + asech(x), x, 0, '+')
log(2)
>>> limit(-log(x) + asech(x), x, 0, '-')
oo
>>> limit(-log(x) + asech(x), x, 0, '+')
oo
>>> limit(log(x) + acsch(x), x, 0, '-')
log(2)
>>> limit(log(x) + acsch(x), x, 0, '+')
log(2)
>>> limit(-log(x) + acsch(x), x, 0, '-')
-log(2) - I*pi
>>> limit(-log(x) + acsch(x), x, 0, '+')
oo
```

Reply:
Pic from wolfram for reference 
![image](https://user-images.githubusercontent.com/87052487/152105175-52f0b6d2-a26b-462d-a8e6-c6ada943f839.png)

Reply:
> It seems that this should be something like `return log(2) - log(c) - e*log(x)` when `arg` is `c*x**e`. If `log(2)` is not included, then the leading term of `log(x) + asech(x)` will not be correct. `log(c*x**e)` should be written in the form `log(c) + e*log(x)` so that the `log(x)` terms coming from different parts of a sum would combine without explicit simplifications.

I think these should be the answers for the following
```
>>> asech(x).as_leading_term(x)
log(2/x)
>>> asech(2*x).as_leading_term(x)
log(1/x)
>>> (2*asech(x)).as_leading_term(x)
2*log(2/x)
>>> ( log(x) + asech(x)).as_leading_term(x)
log(2/x) + log(x)
```
Returning this will give us correct answer for limits !

Reply:
ping @0sidharth

Reply:
Ping @jksuom

Reply:
ping @jksuom @0sidharth